packages/application-shell/src/components/fetch-project/fetch-project.spec.js

```diff
@@ -190,4 +200,83 @@ describe('helpers', () => {
       );
     });
   });
+  describe('mapAllDataFencesToGroupedObjectShape', () => {
+    describe('with store types', () => {
+      const allAppliedDataFences = [
+        {
+          value: 'usa',
+          type: 'store',
+          group: 'orders',
+          name: 'canManageOrders',
+        },
+        {
+          value: 'germany',
+          type: 'store',
+          group: 'orders',
+          name: 'canManageOrders',
+        },
+        {
+          value: 'canada',
+          type: 'store',
+          group: 'orders',
+          name: 'canViewOrders',
+        },
+      ];
+      it('should transform all applied dataFences', () => {
+        expect(
+          mapAllDataFencesToGroupedObjectShape(allAppliedDataFences)
+        ).toEqual({
+          store: {
+            orders: {
+              canManageOrders: {
+                values: ['usa', 'germany'],
+              },
+              canViewOrders: {
+                values: ['canada'],
+              },
+            },
+          },
+        });
+      });
+    });
+    describe('with categories as datafence types', () => {
```

Comment:
We do not support other DataFence types yet in the MC-BE. However this is to test that whatever types we support in the future, we introduce a very common structure in the final output that will look as the following:

```
abstract:
{
  [dataFenceType]: {
    [resourceType]: {
      [accessType]: {
        // dataFenceSpecificValues
        values: []
      }
    }
  }
}

---

real data:
{
  [store]: {
    [orders]: {
      [canManageOrders]: {
        // dataFenceSpecificValues
        values: ['usa', 'germany']
      }
      [canViewOrders]: {
        // dataFenceSpecificValues
        values: ['canada']
      }
    }
  }
}
```

Reply:
Yay. This is dupe.